drivers/SmartThings/matter-switch/src/test/test_matter_switch_device_types.lua

```diff
@@ -217,7 +223,18 @@ local function test_init_onoff_client()
   test.mock_device.add_test_device(mock_device_onoff_client)
 end
 
+local cluster_subscribe_list = {
+  clusters.OnOff.attributes.OnOff
+}
+
 local function test_init_parent_client_child_server()
+  local subscribe_request = cluster_subscribe_list[1]:subscribe(mock_device_parent_client_child_server)
```

Comment:
I am confused why these subscribe requests are needed here when they weren't before? Even before this change, we'd still be sending out a subscribe request for the parent device. Do these tests fail without this added expectation for the subscribe request?

Reply:
There isn't a subscribe request being sent out for the parent device, because in these test cases each mock device starts out with the `matter-thing.yml` profile, so there are not any attributes to subscribe to initially. Now with these changes, for the two mock devices with child devices, when the child devices are created (with actual profiles) we subscribe to the appropriate attributes.